scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+# + addresses - Physical mailing address for the user  
+# + phoneNumbers - Phone numbers for the user  
+# + photos - URI of image of the user  
+# + locale - Default location of the user  
+# + active - Administrative status of the user 
+# + meta - Metadata of the user 
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema  
+# + urn\:scim\:wso2\:schema - Attributes belong to WSO2 custom extension schema  
+# + roles - List of roles for the user
+public type UserResource record {
+    string[]|string schemas?;
+    string id?;
+    string externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    string[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Photo[] photos?;
+    string locale?;
+    boolean active?;
+    Meta meta;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    Custom urn\:scim\:wso2\:schema?;
+    Role[] roles?;
+};
+
+# Represents the sub-attributes of the roles complex attribute in SCIMUser record.
+#
+# + display - Human readable name, primarily used for display purposes
+# + value - Label representing a collection of entitlements  
+# + \$ref - Reference URI 
+# + resourceType - Resource type of the entitlement
+public type Role record {
+    string display;
+    string value?;
+    string \$ref?;
+    string resourceType?;
+};
+
+# Represents the sub-attributes of the meta complex attribute in SCIMUser record.
+#
+# + created - DateTime that the resource was added 
+# + location - URI of the resource being returned 
+# + lastModified - Most recent DateTime that the details of this resource were updated  
+# + resourceType - Name of the resource type of the resource  
+# + 'version - Version of the resource being returned
+public type Meta record {
+    string created;
+    string location;
+    string lastModified;
+    string resourceType?;
+    string 'version?;
+};
+
+# Represents the sub-attributes of the name complex attribute in SCIMUser record.
+#
+# + formatted - Full name, including all middle names, titles, and suffixes as appropriate, formatted for display  
+# + givenName - Given name of the User  
+# + familyName - Family name of the User  
+# + middleName - Middle name(s) of the User 
+# + honorificPrefix - Title 
+# + honorificSuffix - Suffix
+public type Name record {
+    string formatted?;
+    string givenName?;
+    string familyName?;
+    string middleName?;
+    string honorificPrefix?;
+    string honorificSuffix?;
+};
+
+# Represents the Custom Schema 
+#
+# + idpType - Identity provider type  
+# + isReadOnlyUser - Whether the user is read only
+# + userSource - Source of the user 
+# + photoUrl - URI of the user's photo  
+# + userAccountType - Type of the user account  
+# + askPassword - Whether the user is asked to provide a password
+public type Custom record {
```

Comment:
Is this implementation only specific to Asgardeo?
Users can define any attribute under urn:scim:wso2:schema schema